pkg/identity/cache/allocator.go

```diff
@@ -99,6 +102,20 @@ type CachingIdentityAllocator struct {
 	watcher identityWatcher
 
 	owner IdentityAllocatorOwner
+
+	ipc IPCache
+}
+
+// IPCache is an interface hiding the implementation of `pkg/ipcache:IPCache`.
+type IPCache interface {
```

Comment:
I spot four instances of similar (but subtly different) ipcache interfaces here, and it makes me wonder if we should actually centralize the abstraction into some (perhaps independent) package.

Reply:
In these cases, I think if the interface definition is the same, it should be factored out into a separate, shared package. If there are unique combinations of the set of exported methods for the `IPCache` which are only used in one package, I think keeping them localized to said package which uses that interface is sufficient. Thoughts?

Reply:
I think it's more the semantics of the interface rather than the substance of the interface that should define interface reuse.

If the uses are distinct, that makes a lot of sense to me. In this case two of them are the same, then the other two share the same (upsert/delete) functions as the first two but each have one extra function. In my mind there's more to share here than there is to separate. They all are trying to abstract the same notion/semantic of an ipcache, so I think they would benefit from sharing the same interface.

Reply:
PTAL, addressed your concerns